chalk-solve/src/split.rs

```diff
@@ -194,6 +194,146 @@ pub trait Split<I: Interner>: RustIrDatabase<I> {
         let (other_params, trait_params) = parameters.split_at(split_point);
         (trait_params, other_params)
     }
+
+    /// Given the full set of parameters (or binders) for an
```

Comment:
Hmm, this duplication is a bit unfortunate. I wonder if there's a good way to refactor things to avoid it. I guess it's a consequence of trying to preserve the split between associated types and functions.